### PR TITLE
docs: Simplify and update README, CONTRIBUTING and others

### PR DESCRIPTION
**Pull request by jnv**
## Description

- Replaced mentions of "Superface SDK" with "OneSDK" or "Superface OneSDK"
- Simplified README, removed most of the SDK-specific docs in favor of linking to documentation
  - Some specific explanations, like working with results object and environment variables are missing now; I will move them to docs.
- Renamed CONTRIBUTION_GUIDE to CONTRIBUTING since that's file recognized by GitHub
- Updated CONTRIBUTING guide
- Removed ToC since the whole README is short enough; also, broke away from Standard Readme, I don't think it's suitable for projects.
- Added direct links to SF website, docs, Discord, and repeated them a lot in the README just in case someone missed it.

For reference, I was inspired by recommendations and featured projects in [this article](https://skerritt.blog/make-popular-open-source-projects/).

image

[Full README preview](https://github.com/superfaceai/one-sdk-js/blob/docs/readme/README.md)

## Motivation and Context

The README was pretty outdated.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have updated the documentation accordingly. If you are changing **code related to user secrets** you need to really make sure that [security documentation](SECURITY.md) is correct.
- [x] I have read the **CONTRIBUTION_GUIDE** document.
- [x] I haven't repeated the code. (DRY)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,194 +1,122 @@
...
+[Website](https://superface.ai) | [Get Started](https://superface.ai/docs/getting-started) | [Documentation](https://superface.ai/docs) | [Discord](https://sfc.is/discord) | [Twitter](https://twitter.com/superfaceai) | [Support](https://superface.ai/support)
 
-<img src="https://github.com/superfaceai/one-sdk-js/blob/main/docs/LogoGreen.png" alt="superface logo" width="150" height="150">
+<img src="https://github.com/superfaceai/one-sdk-js/raw/main/docs/LogoGreen.png" alt="Superface" width="100" height="100">
 
-Superface is the core SDK of the Superface project. It is the library that communicates with registry and performs operations on profiles/maps, including input/output validations.
+# Superface OneSDK
 
-<!--TODO: Fill out this long description. So, should we fill it :) -->
+**Just one SDK for all the APIs you want to integrate with.**
```
Comment (zdne) [inline README.md:7]: remove "just"

File: README.md [hunk 2]
```diff
@@ -1,194 +1,122 @@
...
-- [Usage](#usage)
-- [Configuration](#configuration)
-- [Metric reporting](#metric-reporting)
-- [Security](#security)
-- [Support](#support)
-- [Maintainers](#maintainers)
-- [Contributing](#contributing)
-- [Licensing](#licensing)
-- [License](#license)
+OneSDK is a universal API client and core library of Superface. It provides a simple and uniform interface across many APIs and automatic runtime switching between API providers.
```
Comment (zdne) [inline README.md:15]: drop "and core library of Superface"

Comment (zdne) [reply README.md:15]: drop "a simple"

Comment (zdne) [reply README.md:15]: "It provides unparalleled developer experience for every HTTP and Graph API" 
"The OneSDK brings unparalleled resiliency to API changes and comes with built-in integration monitoring, testing and fail-overs".

File: README.md [hunk 3]
```diff
@@ -1,194 +1,122 @@
...
 
-## Background
+See [how Superface works](https://superface.ai/how-it-works) and [get started](https://superface.ai/docs/getting-started).
```
Comment (zdne) [inline README.md:17]: "For more details about Superface visit " this and  that

File: README.md [hunk 4]
```diff
@@ -1,194 +1,122 @@
...
-You can get more information at https://superface.ai and https://superface.ai/docs.
+- [Superface website](https://superface.ai)
+- [Get Started](https://superface.ai/docs/getting-started)
+- [Documentation](https://superface.ai/docs)
+- [Discord](https://sfc.is/discord)
 
 ## Install
 
-To install the package, run in the project directory:
+Install the project into Node.js project with npm:
```
Comment (zdne) [inline README.md:28]: 2x project, make it simple like "To install the OneSDK run ..."

File: README.md [hunk 5]
```diff
@@ -1,194 +1,122 @@
...
 ```
 
-#### Performing the use case
+## Usage
 
-**Note**: You can change url of API requests by setting `SUPERFACE_API_URL` environment variable to desired base url.
+💡 **For quick usage example, check [get started](https://superface.ai/docs/getting-started) documentation.**
 
-Make sure a profile is installed by running `superface install <profileName>[@<profileVersion>]` in the project directory, then load the profile:
+OneSDK works with [Superface CLI](https://github.com/superfaceai/cli). First you install a profile for the use-case you want to use by executing this command in the project directory:
```
Comment (zdne) [inline README.md:44]: "With OneSDK everything revolves about your application's use cases for an API. To get started, first install a use case profile using the superface CLI ..."

File: README.md [hunk 6]
```diff
@@ -1,194 +1,122 @@
...
 
-```typescript
-const profile = await client.getProfile('<profileName>');
+```shell
+npx @superfaceai/cli install <profileName>
 ```
 
-Next, make sure at least one provider is configured in super.json or select one manually. You can configure providers in super.json by running `superface configure <providerName>` and you can add additional or overriding configuration by calling `.configure` on the Provider object:
+Next you configure a provider for the use-case, optionally with API keys:
```
Comment (zdne) [inline README.md:50]: Drop "optionally with API keys" add note on API keys later (maybe right after?)

File: README.md [hunk 7]
```diff
@@ -1,194 +1,122 @@
...
-    inputField: 1,
-    anotherInputField: 'hello',
-  },
-  // optional, if provider is missing selects first configured provider from super.json
-  { provider }
-);
-```
-
-### Handling the results from `perform`
+The CLI creates a configuration file in `superface/super.json` file which is loaded by OneSDK.
```
Comment (zdne) [inline README.md:56]: Drop "which is loaded by OneSDK."

File: README.md [hunk 8]
```diff
@@ -1,194 +1,122 @@
...
-```typescript
-result.match(
-  value => console.log(value),
-  error => console.log(error.toString())
-);
+run();
 ```
 
-#### Unsafely unwrapping the result
+Optionally you can [define which provider should be used](https://superface.ai/docs/guides/using-multiple-providers).
```
Comment (zdne) [inline README.md:78]: Change it to "This code will use the provider specified in the super.json. You can override the configure in the code like this and that" 


----

bottom line: I feel you are referring too much to external docs. Ideally, at least the basic useage should be completely covered by the README.

File: README.md [hunk 9]
```diff
@@ -1,194 +1,122 @@
...
-try {
-  // Possible error is thrown here and it contains human readable description of what went wrong :)
-  const value = result.unwrap();
-  // You can accees value here
-  console.log(value);
-} catch (e) {
-  console.log(e);
-}
-```
+Check the [Superface Catalog](https://superface.ai/catalog) for existing use-cases or learn how to [write your own use-cases](https://superface.ai/docs/guides/how-to-create).
```
Comment (zdne) [inline README.md:80]: "To find available use-case, check the <link to station> or sign up for Superface and visit its use-cases catalog"

Comment (zdne) [reply README.md:80]: "If you are missing a use case, let us know <link to contacts>. You can always add your own use case or API provider <link to docs>"

File: README.md [hunk 10]
```diff
@@ -1,194 +1,122 @@
...
-There are three kinds of metrics reported at present - one is sent when the client instance is created, one after each perform (reporting success or failure) and one when a provider failover happens. The reports can be disabled or configured with [environment variables](#configuration).
-For metrics to be successfuly sent, the application needs to be properly exited, i.e. there should be no unhandled Promise rejections or exceptions.
+You can find more options for reaching us on the [Support page](https://superface.ai/support).
 
 ## Security
 
-Superface is not man-in-the-middle so it does not require any access to secrets that are needed to communicate with provider API. Superface SDK only reads super.json file, resolved authorization secrets from environment variables or from the file itself and applies them to network requests as required by the specific map.
+Superface doesn't act as a proxy and communicates with API providers directly; the secrets are sent only to the providers (see [how Superface works](https://superface.ai/how-it-works)).
```
Comment (zdne) [inline README.md:92]: Superface is not a proxy. The calls are always going directly from your application to API providers. The API secrets are never sent anywhere else but to the used provider's API ...

File: SECURITY.md [hunk 11]
```diff
@@ -1,6 +1,6 @@
-# Security within Superface SDK
+# Security within Superface OneSDK
 
-The Superface SDK interprets the map to fulfill the required usecase. This often requires making http requests with authorization. Thus the user needs to provide Superface with the right secrets to access the capability exposed by the API.
+OneSDK interprets the map to fulfill the required usecase. This often requires making http requests with authorization. Thus the user needs to provide Superface with the right secrets to access the capability exposed by the API.
```
Comment (zdne) [inline SECURITY.md:3]: I do not understand what this is saying and actually it is raising some concerns in my eyes... maybe rephrase it?

File: SECURITY.md [hunk 12]
```diff
@@ -45,4 +45,4 @@ Once the request is executed no secrets are accessed until another request is to
 
 Another aspect to consider is logging. Logging is **disabled** by default. When enabled, logging may leak user secrets, so an appropriate logging level should be selected to only expose as much information as is secure in given context.
```
Comment (zdne) [inline SECURITY.md:46]: Jirasek :D Drop "Another aspect to consider is logging"

**Review by zdne — CHANGES_REQUESTED** [review]
I love it! Much much better. I'd still add one or two lines on "benefits" why you want to use it instead of anything else...

File: SECURITY.md (hunk 12 above)
Comment (jnv) [reply SECURITY.md:46]: Heh, I haven't really checked the SECURITY document, only to fix "Superface SDK". I would like to go through it in a separate PR, ok?

File: README.md (hunk 2 above)
Comment (jnv) [reply README.md:15]: Adapted like this:

> OneSDK is a universal API client which provides an unparalleled developer experience for every HTTP and Graph API. It enhances resiliency to API changes and comes with built-in integration monitoring, testing and fail-overs.

File: README.md (hunk 6 above)
Comment (jnv) [reply README.md:50]: Added this paragraph after the command:

> CLI may instruct you about setting up API keys if the provider needs them.

**Comment by jnv** [comment]
@zdne Thanks for the feedback! I have added your suggestions beside the SECURITY which I'd like to tackle separately.

I'd like to hear from SDK authors, especially @lukas-valenta, @Jakub-Vacek, and @TheEdward162. Any ideas / suggestions?

**Comment by jnv** [comment]
> I love it! Much much better. I'd still add one or two lines on "benefits" why you want to use it instead of anything else...

@zdne I thought about adding a "Features" section with the list of benefits we shared in early access email:

> - One simple to use Node.js SDK for all the APIs you want to use
> - No need to read API docs
> - Immunity to API changes
> - Integration monitoring and testing
> - Clean, hexagonal architecture for your integrations
> - Platform-less integration, harmonization and aggregation without a middleman
> - Reduction of your application's code base
> - The super power for your app to self-integrate
> - Automatic fail-overs*
> - No vendor locks*

However, most of these apply to Superface as a whole and I feel it's kinda misleading to attribute these features to OneSDK. For example, "no need to read API docs" and "immunity to API changes" applies only if you are not building the use-cases yourself, but on the other hand this is one of the benefits of OneSDK.

But if you'd like to emphasise these features, maybe we could add this:

----

## Features

- One Node.js SDK for all the APIs you want to use
- Automatic fail-overs
- Built-in support for integration monitoring
- Clean, hexagonal architecture for integrations
- Platform-less integration, harmonization, and aggregation without a middleman
- Reduction of your application's code base

----

Maybe also add something like "full control over integrations' code" or somehow underline the customizability of integrations.

**Review by zdne — COMMENTED** [review]
I do not want to be a blocker on this, @jnv please proceed as you think is the best.
